merlin/dataloader/tensorflow.py

```diff
@@ -88,7 +88,11 @@ def __next__(self):
     def peek(self):
         """Grab the next batch from the dataloader
         without removing it from the queue"""
-        return self.convert_batch(self._peek_next_batch())
+        converted_batch = self.convert_batch(self._peek_next_batch())
+        for map_fn in self._map_fns:
```

Comment:
looks like the same applies to the peek method of the torch dataloader too. Would it work to move these methods (`__next__`/`peek`) to the base class like we had before?

Reply:
either in this PR or a new one.

Reply:
Thanks, that's a good point. I'll follow up with a new PR.